src/FSharp.Data.GraphQL.Server/Execution.fs

```diff
@@ -368,13 +439,74 @@ let internal executePlan (ctx: ExecutionContext) (plan: ExecutionPlan) (objdef:
+                        Context = ctx
+                        ReturnType = fdef.TypeDef
+                        ParentType = objdef
+                        Schema = ctx.Schema
+                        Args = args
+                        Variables = ctx.Variables
+                    } 
+                    let rec traversePath (path: string list) (tree: AsyncVal<ResolverTree>) (pathAcc: string list): AsyncVal<(ResolverTree * string list) []> =
+                        tree 
+                        |> AsyncVal.bind(fun tree' ->
+                            match List.tail path, tree' with
+                            | [], t -> AsyncVal.wrap([|buildResolverTree info.ReturnDef fieldCtx fieldExecuteMap t.Value, List.rev ((List.head path)::pathAcc)|])
+                            | [p], t -> AsyncVal.wrap([|buildResolverTree info.ReturnDef fieldCtx fieldExecuteMap t.Value, List.rev(p::pathAcc)|])
+                            | [p;"__index"], t -> AsyncVal.wrap([|buildResolverTree info.ReturnDef fieldCtx fieldExecuteMap t.Value, List.rev(p::pathAcc)|])
+                            | p, ResolverObjectNode n -> 
+                                n.Children 
+                                |> AsyncVal.bind(fun c ->
+                                    let next = c |> Array.tryFind(fun c' -> c'.Name = (List.head p))
+                                    match next with
+                                    | Some next' -> traversePath p (AsyncVal.wrap next') ((List.head p)::pathAcc)
+                                    | None -> AsyncVal.empty)
+                            | p, ResolverListNode l ->
+                                l.Children 
+                                |> AsyncVal.bind(fun c ->
+                                    c
+                                    |> Array.mapi(fun i c' -> 
+                                        traversePath p (AsyncVal.wrap c') (i.ToString()::pathAcc)
+                                    )
+                                    |> AsyncVal.collectParallel
+                                    |> AsyncVal.map(Array.fold (Array.append) [||]))
+                            | _ ,_ -> raise <| GraphQLException("Deferred path terminated unexpectedly!"))
+                    traversePath path (AsyncVal.wrap tree) [(List.head path)]
+                    |> AsyncVal.bind(Array.map(fun (tree, path) ->
+                        asyncVal {
+                            let! tree' = treeToDict tree
+                            return NameValueLookup.ofList["data", upcast tree'; "path", upcast path]
+                        }) >> AsyncVal.collectParallel))
+                |> AsyncVal.collectParallel
+                |> AsyncVal.map(Array.fold (Array.append) [||])))
+            |> AsyncVal.collectParallel
+            |> AsyncVal.map(Array.fold (Array.append) [||])
+            |> AsyncVal.toAsync
+            |> Observable.ofAsync
+            |> Observable.bind(Observable.ofSeq)
```

Comment:
If I read it right, you're basically load a whole collection to be deferred into memory, and `@defer` is just used to return it in chunks as `IObservable<>`? If it's true, I don't see any value added by defer directive.